paddle/math/Matrix.h

```diff
@@ -370,7 +370,7 @@ class Matrix : public BaseMatrix {
    * allocate matTrans' memory outside, then set memAlloc as false;
    * else set as true.
    */
-  virtual void transpose(MatrixPtr matTrans, bool memAlloc) {
+  virtual void transpose(MatrixPtr& matTrans, bool memAlloc) {
```

Comment:
I think the better implementation is:
void transpose(Matrix& src) {
  transpose the src and assign  to this;
}
By the way, `transpose(dst, true)` does not seem to be necessary.